ocfweb/api/hours.py

```diff
@@ -0,0 +1,104 @@
+from collections import namedtuple
+from datetime import date
+from datetime import datetime
+from datetime import time
+from datetime import timedelta
+
+import requests
+from django.http import JsonResponse
+
+from ocfweb.caching import periodic
+
+HOURS_SPREADSHEET = 'https://docs.google.com/spreadsheet/ccc?key=1WgczUrxqey63fmPRmdkCDMCbsfaTyCJrixiCeJj35UI&output=csv'  # noqa: E501
+SHIFT_LENGTH = timedelta(hours=0.5)
+
+
+class Hour(namedtuple('Hours', ['open', 'close', 'staffer'])):
+
+    def __contains__(self, when):
+        if not isinstance(when, time):
+            when = when.time()
+
+        return self.open <= when < self.close
+
+
+def _get_hours():
+    """pull hours from the Google spreadsheet"""
+
+    response = requests.get(HOURS_SPREADSHEET)
+    response.raise_for_status()
+
+    matrix = response.content.decode('utf-8').splitlines()
+    matrix = [row.split(',') for row in matrix]
+
+    # [1:] because the first box in the matrix is empty
+    # matrix[0] = ['', 'Monday', 'Tuesday', ...]
+    # row[0] = ['13:30PM', 'name1', 'name2', ...]
+    shifts = [row[0] for row in matrix][1:]
```

Comment:
assert this in case it changes? you can replace this whole thing with

```python
# The first row is a header and has no time slot
first_row = shifts.pop(0)
assert first_row[0] == '', first_row
```